src/linkediting.js

```diff
@@ -187,6 +189,41 @@ export default class LinkEditing extends Plugin {
 					key: decorator.id
 				}
 			} );
+
+			// For a decorator that is enabled by default we must check if it was applied to reflect that in the model.
+			if ( decorator.defaultValue ) {
+				editor.conversion.for( 'upcast' ).add( dispatcher => {
+					const matcher = new Matcher( {
+						name: 'a',
+						attributes: {
+							href: true
+						}
+					} );
+
+					dispatcher.on( 'element', ( evt, data, conversionApi ) => {
+						const matcherResult = matcher.match( data.viewItem );
+
+						if ( !matcherResult ) {
+							return;
+						}
+
+						// We must check if all decorator values are set on view element.
+						const decoratorValue = Object.entries( decorator.attributes ).every( ( [ key, value ] ) => {
```

Comment:
I wanted to write that this comparison is not precise enough because you can have two decorators using the class attribute but I can see that this is not supported already. It's a pity because the engine treats each class separately (styles and classes are special types of attributes) and this could work. In fact, getData() works, but setData() breaks.

Reply:
I reported a ticket for that in ckeditor/ckeditor5#6571.